cmd/thanos/flags.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/cluster"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func registerServerFlags(cmd *kingpin.CmdClause) (grpcBindAddr *string, httpBindAddr *string, newPeerFn func(log.Logger, *prometheus.Registry, bool, string, bool) (*cluster.Peer, error)) {
+	grpcBindAddr = cmd.Flag("grpc-address", "Listen ip:port address for gRPC endpoints (StoreAPI). Make sure this address is routable from other components if you use gossip, 'grpc-advertise-address' is empty and you require cross-node connection.").
+		Default("0.0.0.0:10901").String()
```

Comment:
Think i still prefer constants here ... and will help with `compact.go` setting up with the same addr as well.

Reply:
No more string in `compact.go`. So it is only used here.